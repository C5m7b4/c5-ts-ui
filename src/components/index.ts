import {
  Button,
  NormalButtonType,
  NormalButtonSize,
  ButtonProps,
} from './Button';

export { Button, NormalButtonType, NormalButtonSize, ButtonProps };
