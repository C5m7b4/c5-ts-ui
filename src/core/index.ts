import Backdrop, { BackdropProps } from './Backdrop';
import Modal, { ModalProps } from './Modal';
import ModalWrapper, { ModalWrapperProps } from './ModalWrapper';

export {
  Backdrop,
  BackdropProps,
  Modal,
  ModalProps,
  ModalWrapper,
  ModalWrapperProps,
};
