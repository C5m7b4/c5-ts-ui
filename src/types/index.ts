export type Theme = 'light' | 'dark' | 'colored';

export type Parsers =
  | 'babel'
  | 'css'
  | 'json'
  | 'flow'
  | 'babel-flow'
  | 'babel-ts'
  | 'typescript'
  | 'json'
  | 'markdown'
  | 'html';
