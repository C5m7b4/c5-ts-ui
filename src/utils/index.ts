export * from './colors';
export * from './reserved';
export * from './fns';
