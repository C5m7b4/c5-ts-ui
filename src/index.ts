import { Add } from './math';
import {
  Button,
  NormalButtonType,
  NormalButtonSize,
  ButtonProps,
  SelectField,
  SelectFieldProps,
  SelectTheme,
  TextTheme,
  TextFieldType,
  TextFieldAutoComplete,
  TextFieldProps,
  TextField,
  RippleTypes,
  RippleButtonProps,
  RippleButton,
  OutlineButton,
  OutlineButtonProps,
  OutlineButtonTypes,
} from './components';

import { Theme } from './types';

export {
  Add,
  Button,
  NormalButtonType,
  NormalButtonSize,
  ButtonProps,
  Theme,
  SelectField,
  SelectFieldProps,
  SelectTheme,
  TextTheme,
  TextFieldType,
  TextFieldAutoComplete,
  TextFieldProps,
  TextField,
  RippleTypes,
  RippleButtonProps,
  RippleButton,
  OutlineButton,
  OutlineButtonProps,
  OutlineButtonTypes,
};
