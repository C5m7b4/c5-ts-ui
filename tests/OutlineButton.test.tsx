import React from 'react';
import { render } from '@testing-library/react';

import { OutlineButton } from '../src';

describe('OutlineButton tests', () => {
  test('should render correctly', () => {
    const testFn = jest.fn;
    const { container } = render(
      <OutlineButton text="hello" type="default" onClick={testFn} />
    );
    expect(container).toMatchSnapshot();
  });
});
