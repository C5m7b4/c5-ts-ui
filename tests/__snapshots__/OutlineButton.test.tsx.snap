// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutlineButton tests should render correctly 1`] = `
<div>
  <button
    class="c5-btn default"
  >
    hello
  </button>
</div>
`;
